File: dotnet/Gwhr.ResourceGenerator/Models/GhResourceDocument.cs
namespace Gwhr.ResourceGenerator.Models;

public class GhResourceDocument
{
    public string Name { get; set; } = string.Empty;
    public string DirectoryPath { get; set; } = string.Empty;
    public Dictionary<string, GhResourceItem> Items = new Dictionary<string, GhResourceItem>();
}

File: resgen-ts/src/models/ghResourceDocument.js
export class GhResourceDocument {
    constructor(name, directoryPath) {
        this.name = "";
        this.directoryPath = "";
        this.items = new Map();
        this.name = name == null ? this.name : name;
        this.directoryPath = directoryPath == null ? this.directoryPath : directoryPath;
    }
}
